kernelci/configs.py

```diff
@@ -262,9 +262,10 @@ def from_yaml(cls, data, name, fragments):
         kw.update(cls._kw_from_yaml(data, [
             'name', 'base_defconfig', 'extra_configs',
         ]))
-        cf = data.get('fragments')
-        kw['fragments'] = [fragments[name] for name in cf] if cf else None
-        kw['filters'] = FilterFactory.from_data(data)
+        if data is not None:
```

Comment:
It would be better to not call this method with data being `None` rather than checking its value.  I think it's important to draw the line with what the values for the methods should be, I guess that would require some docstrings in the `from_yaml()` methods.

So how about this patch instead, to fix it on the calling side:
```diff
diff --git a/kernelci/configs.py b/kernelci/configs.py
index 42b4c2de9da9..87ecfb863d9b 100644
--- a/kernelci/configs.py
+++ b/kernelci/configs.py
@@ -373,7 +373,7 @@ class BuildVariant(YAMLObject):
             config, ['name', 'build_environment', 'fragments']))
         kw['build_environment'] = build_environments[kw['build_environment']]
         kw['architectures'] = list(
-            Architecture.from_yaml(data, name, fragments)
+            Architecture.from_yaml(data or {}, name, fragments)
             for name, data in config['architectures'].iteritems()
         )
         cf = kw.get('fragments')
```

Reply:
Thanks, that is better.